proj/src/main/webapp/chat.js

```diff
@@ -531,11 +534,69 @@ function createMessageWithTemplate(key, messageObj) {
     timestamp.innerText = new Date(messageObj.timestamp).toLocaleString();
 
     message.id = key;
+    message.dataset.userId = messageObj.senderUID;
     addUsernameToMessage(messageObj.senderUID, message)
     return message;
 }
 
+// onclick for messages
+function loadProfileOfSender(domElement, uid) {
+    domElement.addEventListener('click', function() {
+        populateProfileSidebar(uid);
+    })
+}
 
+function friendRequestButton(uid) {
+    const currUserRef = firebase.database().ref('/users/'+currentUID+'/friend-requests/'+uid);
+    const otherUserRef = firebase.database().ref('/users/'+uid+'/friend-requests/'+currentUID);
+
+    const button = document.getElementById('friend-request');
+
+    currUserRef.off();
+    currUserRef.on('value', function(snap) {
+        switch (snap.val()) {
+            case 'sent':
+                // cancel friend request
+                button.innerText = 'cancel friend request';
+                button.onclick = function() {
+                    currUserRef.remove();
+                    otherUserRef.remove();
+                };
+                break;
+            case 'received':
+                // accept or deny
+                button.innerText = 'accept friend request';
+                button.onclick = function() {
+                    currUserRef.remove();
+                    otherUserRef.remove();
+
+                    const currFriendRef = firebase.database().ref('/users/'+currentUID+'/friends/'+uid);
+                    const otherFriendRef = firebase.database().ref('/users/'+uid+'/friends/'+currentUID);
+                    currFriendRef.set(true);
+                    otherFriendRef.set(true);
+                };
```

Comment:
deny should also be hidden if they click on accept. Can you add a `deny.hidden=true;` to this block.

Reply:
fixed